src/Neo/Network/P2P/LocalNode.cs

```diff
@@ -263,6 +265,25 @@ private void OnRelayDirectly(IInventory inventory)
                 SendToRemoteNodes(message);
         }
 
+        public NodeCapability[] GetNodeCapabilities()
+        {
+            var capabilities = new List<NodeCapability>
+            {
+                new FullNodeCapability(NativeContract.Ledger.CurrentIndex(system.StoreView))
+                // Wait for 3.9
+                // new ArchivalNodeCapability()
```

Comment:
To me, it shouldn't be commented out because all C# nodes are archival in fact (there's no tails cutting functionality).

Reply:
> We can't do this in 3.8, old nodes will disconnect immediately. This was deliberately postponed to 3.9.